docker/quick-start/bacalhau-entrypoint.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+SUDO='' # detect if not root...
+
+if [[ -z "${IPFS_CONNECT:-}"  ]]; then
+  # if bacalhau-ipfs container isn't running, start one
+  if ! docker inspect bacalhau-ipfs > /dev/null ; then
+    # TODO: can we put the desired version into the bacalhau binary?
+    export IPFS_VERSION=$(wget -q -O - https://raw.githubusercontent.com/filecoin-project/bacalhau/main/ops/terraform/production.tfvars | grep --color=never ipfs_version | awk -F'"' '{print $2}')
+    #ignoring the repo version - they're all too old atm
+    export IPFS_VERSION=latest
+    # run IPFS (this should eventually be run in the bacalhau entrypoint)
+    # TODO: run container as readonly fs
+
+    echo "Starting the backalhau-ipfs ${IPFS_VERSION} container"
+    docker run -d \
+        --restart always \
+        --name bacalhau-ipfs \
+        -v bacalhau_ipfs_staging:/export \
+        -v bacalhau_ipfs_data:/data/ipfs \
+        -p 4001:4001 \
+        -p 4001:4001/udp \
+        -p 127.0.0.1:8080:8080 \
+        -p 127.0.0.1:5001:5001 \
+            ipfs/kubo:${IPFS_VERSION}
+
+    # TODO: how do we test if the image has pulled, and we're started enough to continue?
```

Comment:
Yeah exactly, not great.

Reply:
grin - that's why the RancherOS installer container pattern became the other `docker run -it -v dockersocket complexservice:installer` which then could do the interactive things a user needed to answer, and finish by running `docker compose up` - and exit.

Reply:
humm, looking into what `bacalhau devstack` does - it's also starting dependant containers from what will be the bacalhau container - for example the `LOTUS_TEST_IMAGE`